integration-tests/bats/sql-merge.bats

```diff
@@ -1130,6 +1130,173 @@ SQL
+    dolt SQL -q "INSERT INTO t2 values (9)"
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "t2" ]] || false
+    [[ ! "$output" =~ "test1" ]] || false
+
+    run dolt sql -q "CALL DOLT_MERGE('--squash', 'merge_branch', '--no-commit');"
+    log_status_eq 0
+    [[ "$output" =~ "| 0            | 0         |" ]] || false
+
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "t2" ]] || false
+    [[ "$output" =~ "test" ]] || false
+
+    # make sure the squashed commit is not in the log.
+    dolt add .
+    dolt commit -m "squash merge"
+
+    run dolt log
+    log_status_eq 0
+    [[ "$output" =~ "add pk 4 to test" ]] || false
+    [[ ! "$output" =~ "add pk 3 to test" ]] || false
+}
+
+@test "sql-merge: --abort restores working changes" {
+    dolt commit -m "initial commit"
+    dolt branch other
+
+    dolt sql -q "INSERT INTO test VALUES (9),(8);"
+    dolt commit -am "added rows to test on main"
+
+    dolt checkout other
+    dolt sql -q "INSERT INTO test VALUES (7),(6);"
+    dolt commit -am "added rows to test on other"
+
+    dolt checkout main
+    # dirty the working set with changes to test2
+    dolt SQL -q "CREATE TABLE t2 (pk int)"
+    dolt sql -q "INSERT INTO t2 VALUES (5);"
+
+    dolt merge other --no-commit
+    dolt sql -q "call dolt_merge('--abort');"
+
+    run dolt sql -q "SELECT * from dolt_merge_status"
+    [[ "$output" =~ "false" ]] || false
+
+    # per Git, working set changes to test2 should remain
+    dolt sql -q "SELECT * FROM t2" -r csv
+    run dolt sql -q "SELECT * FROM t2" -r csv
+    log_status_eq 0
+    [[ "${lines[1]}" =~ "5" ]] || false
+}
+
+@test "sql-merge: 3way merge doesn't stomp working changes" {
+    dolt commit -m "initial commit"
+    dolt checkout -b merge_branch
+    dolt SQL -q "INSERT INTO test values (9)"
+    dolt add test
+    dolt commit -m "add pk 9 to test1"
+
+    dolt checkout main
+    dolt SQL -q "INSERT INTO test values (8)"
+    dolt add test
+    dolt commit -m "add pk 8 to test1"
+
+    dolt SQL -q "CREATE TABLE t2 (pk int)"
+    dolt SQL -q "INSERT INTO t2 values (7)"
+    run dolt status
+    log_status_eq 0
+    [[ "$output" =~ "t2" ]] || false
+    [[ ! "$output" =~ "test" ]] || false
+
+    run dolt sql -q "CALL DOLT_MERGE('merge_branch', '--no-commit');"
+    log_status_eq 0
+    [[ ! "$output" =~ "| 1            |" ]] || false
+
+    run dolt status
+    echo -e "\n\noutput: " $output "\n\n"
+    log_status_eq 0
+    [[ "$output" =~ "t2" ]] || false
+    [[ "$output" =~ "test" ]] || false
+
+   run dolt sql -q "SELECT * from dolt_merge_status"
+   [[ "$output" =~ "true" ]] || false
+   [[ "$output" =~ "merge_branch" ]] || false
+   [[ "$output" =~ "refs/heads/main" ]] || false
+
+    # make sure all the commits make it into the log
+    dolt add .
+    dolt commit -m "squash merge"
+
+    run dolt log
+    log_status_eq 0
+    [[ "$output" =~ "add pk 9 to test" ]] || false
+    [[ "$output" =~ "add pk 8 to test" ]] || false
+}
+
```

Comment:
I don't see any tests for fast forward merges with a dirty directory. Looks like you changed the behavior of the executeFFMerge method, and I'd expect that to need a new test or break an old test... or is this covered because several old tests we moved to the sql call rather than `dolt merge`?